scripts/tablegen.py

```diff
@@ -230,35 +237,224 @@ def sanitizeValueStr(value):
+# k - istio component name for which these configuration options are being processed. This is
+#     used to populate the contents of 'od_diff' dictionary.
+#
+def compareValues(oValue, nValue, k):
+    # oValue and nVAlue contains configuration option in the format: 
+    # '| `<Key>` | `<Value>` | `<Description>` |
+    # This needs to be split in order to get the Key, Value and Description values to compare.
+    oldKey = ''
+    oldValue = ''
+    oldDesc = ''
+    
+    newKey = ''
+    newValue = ''
+    newDesc = ''
+            
+    if oValue is not None:
+        groups = re.search("\| \`(.*)\` \| \`(.*)\` \|\s*(.*)\s*\|", oValue.strip())
+        
+        if groups:
+            oldKey = groups.group(1)
+            oldValue = groups.group(2)
+            oldDesc = groups.group(3)
+            
+    if nValue is not None:   
+        groups = re.search("\| \`(.*)\` \| \`(.*)\` \| (.*) |", nValue.strip())
+        if groups:
+            newKey = groups.group(1)
+            newValue = groups.group(2)
+            newDesc = groups.group(3)
+        
+    # if the configuration option values in both versions are the same, then we
+    # can ignore it. We only want to track the differences.
+    if oValue is not None and nValue is not None and oValue == nValue:
+        return oldKey
+    elif oValue is not None and nValue is not None and oValue != nValue:
+        if oldKey != newKey:
+            # if this is a newKey being added in the latest version, for which
+            # a corresponding key does not exist in the previous version, we will
+            # 'n/a' to represent the oldKey. This is true vice-versa and for
+            # comparisons of the 'Value' and 'Description' fields as well ('n/a'
+            # will be used to represent the fact that there is no corresponding
+            # value). 
+            if oldKey is None:
+                oldKey = 'n/a'
+            if newKey is None:
+                newKey = 'n/a'
+            
+        if oldValue != newValue:
+            if oldValue is None:
+                oldValue = 'n/a'
+            if newValue is None:
+                newValue = 'n/a'
+            
+        if oldDesc != newDesc:
+            if (newDesc == None or newDesc == '') and (oldDesc is None or oldDesc == ''):
+                pass
+    
+            if oldDesc is None:
+                oldDesc = 'n/a'
+            if newDesc is None or newDesc == '':
+                newDesc = 'n/a'
+            
+    elif oValue is None:
+            oldKey = 'n/a'
+            oldValue = 'n/a'
+            oldDesc = 'n/a'
+    elif nValue is None:
+            newKey = 'n/a'
+            newValue = 'n/a'
+            newDesc = 'n/a'
+            
+    od_diff[k].append("| `%s` | `%s` | `%s` | `%s` | `%s` | `%s` |" % (oldKey, newKey, oldValue.rstrip(), newValue.rstrip(), oldDesc, newDesc))
+    
+    return oldKey
+        
+def getIndexDiffMeta(prevRelease):
+    meta = ("---\n"
```

Comment:
I would recommend using the pattern laid out in the other generated docs where the front matter is not generated, but isntead the tables are replaced.  This permits people that are not  familiar with python or how this tool works to modify the content for readability.

Reply:
Done. I never really understood why we were not generating the front matter in the first place but now it makes sense. Thanks for explaining @sdake